akka-typed/src/main/scala/akka/typed/ScalaProcess.scala

```diff
@@ -0,0 +1,538 @@
+     * Wrap as a [[Process]] with the given mailbox capacity and infinite timeout.
+     */
+    def withMailboxCapacity(mailboxCapacity: Int): Process[S, Out] = named("").withMailboxCapacity(mailboxCapacity)
+
+    /**
+     * Wrap as a [[Process]] with the given timeout and a mailbox capacity of 1.
+     */
+    def withTimeout(timeout: Duration): Process[S, Out] = named("").withTimeout(timeout)
+
+    /**
+     * Wrap as a [[Process]] but without a name and convert to a [[Behavior]].
+     */
+    def toBehavior: Behavior[ActorCmd[S]] = named("").toBehavior
+
+  }
+
+  /*
+   * These are the private values that make up the core algebra.
+   */
+
+  private[typed] case class FlatMap[S, Out1, Out2](first: Operation[S, Out1], then: Out1 ⇒ Operation[S, Out2]) extends Operation[S, Out2] {
+    override def toString: String = s"FlatMap($first)"
+  }
+  private[typed] case object ShortCircuit extends Operation[Nothing, Nothing] {
+    override def flatMap[T](f: Nothing ⇒ Operation[Nothing, T]): Operation[Nothing, T] = this
+  }
+
+  private[typed] case object System extends Operation[Nothing, ActorSystem[Nothing]]
+  private[typed] case object Read extends Operation[Nothing, Nothing]
+  private[typed] case object ProcessSelf extends Operation[Nothing, ActorRef[Any]]
+  private[typed] case object ActorSelf extends Operation[Nothing, ActorRef[ActorCmd[Nothing]]]
+  private[typed] case class Return[T](value: T) extends Operation[Nothing, T]
+  private[typed] case class Call[S, T](process: Process[S, T]) extends Operation[Nothing, T]
+  private[typed] case class Fork[S](process: Process[S, Any]) extends Operation[Nothing, SubActor[S]]
+  private[typed] case class Spawn[S](process: Process[S, Any], deployment: DeploymentConfig) extends Operation[Nothing, ActorRef[ActorCmd[S]]]
+  private[typed] case class Schedule[T](delay: FiniteDuration, msg: T, target: ActorRef[T]) extends Operation[Nothing, a.Cancellable]
+  private[typed] sealed trait AbstractWatchRef { type Msg }
+  private[typed] case class WatchRef[T](watchee: ActorRef[Nothing], msg: T, target: ActorRef[T])
+    extends Operation[Nothing, a.Cancellable] with AbstractWatchRef {
+    type Msg = T
+    override def equals(other: Any) = super.equals(other)
+    override def hashCode() = super.hashCode()
+  }
+  private[typed] case class Replay[T](key: StateKey[T]) extends Operation[Nothing, T]
+  private[typed] case class Snapshot[T](key: StateKey[T]) extends Operation[Nothing, T]
+  private[typed] case class State[S, T <: StateKey[S], E](key: T, afterUpdates: Boolean, transform: S ⇒ (Seq[T#Event], E)) extends Operation[Nothing, E]
+  private[typed] case class StateR[S, T <: StateKey[S]](key: T, afterUpdates: Boolean, transform: S ⇒ Seq[T#Event]) extends Operation[Nothing, S]
+  private[typed] case class Forget[T](key: StateKey[T]) extends Operation[Nothing, akka.Done]
+
+  // FIXME figure out cleanup of external resources after a failure
+
+  /*
+   * The core operations: keep these minimal!
+   */
```

Comment:
I fear there will have to be some bikeshed about the`op` prefix there... I see why with prefix and not plain, since people will want to call the values `system` for example etc, but perhaps we can get away with something nicer.
E.g. why not keep the UpperCase directly for the ops?

```
for { 
  system <- System
  self <- ActorSelf
] 
```

not too bad?